docs/typescript/reference/dbos-class.md

```diff
@@ -0,0 +1,957 @@
+```
+'sunday',    'sun' -> 0
+'monday',    'mon' -> 1
+'tuesday',   'tue' -> 2
+'wednesday', 'wed' -> 3
+'thursday',  'thu' -> 4
+'friday',    'fri' -> 5
+'saturday'   'sat' -> 6
+```
+
+#### Matching
+For a scheduled workflow to run at a given time, the time must match the crontab pattern.
+A time matches the pattern if all fields of the time match the pattern.
+Each field matches the pattern if its numerical value is within any of the inclusive ranges provided in the field, and is also divisible by the divisor.
+
+## Application Lifecycle
+For DBOS applications using entrypoint classes specified in [`dbos-config.yaml`](../reference/configuration), a "serverless" application lifecycle is handled automatically.  However, it is possible to build a custom application lifecycle.
+
+### Setting The Application Configuration
+If the DBOS configuration should not be loaded from `dbos-config.yaml`, it may be provided programatically with `DBOS.setConfig`:
+```typescript
+DBOS.setConfig(config: DBOSConfig, runtimeConfig?: DBOSRuntimeConfig)
+```
+
+If `runtimeConfig` is provided, the runtime will be started when the app is launched with `DBOS.launch`, including HTTP handling for decorated methods.  If `runtimeConfig` is not provided, this will not occur.
+
+### Launching and Shutting Down
+The DBOS app can be started with `launch` and stopped with `shutdown`.
+```typescript
+DBOS.launch(httpApps?: DBOSHttpApps)
+DBOS.shutdown()
+```
+
+* `launch`: This starts the DBOS app, or registers it to be started in conjunction with a provided HTTP server.  Launching will initialize all DBOS components, run any application initializers, start workflow recovery, start event processing, and start HTTP services (if so configured).
+* `shutdown`: This stops the DBOS app.  First, handling of new events and requests is disabled, and then the workflow executor and database connections are destroyed.
+
+### Recovering Workflows
+DBOS generally recovers workflows automatically.  However, this process can be initiated externally.
+
+```typescript
+DBOS.recoverPendingWorkflows(executorIDs: string[] = ["local"]): Promise<WorkflowHandle<unknown>[]>
+DBOS.executeWorkflowById(workflowId: string, startNewWorkflow: boolean = false): Promise<WorkflowHandle<unknown>>
+```
+
+* `DBOS.recoverPendingWorkflows`: Retrieves a list of workflows to recover, and starts them, returning their handles
+* `DBOS.executeWorkflowById`: Starts executing a workflow given its `workflowId`.  If `startNewWorkflow` is `true`, the workflow is cloned and assigned a new unique workflow ID.
+
+## Extensibility
+DBOS Transact allows libraries, such as those designed to receive external events, to initiate DBOS functions.
+
+### Decorators
+Such libraries may choose to implement method decorators.  If the method decorator is registering a function with DBOS Transact, it should call `DBOS.registerAndWrapContextFreeFunction` for functions that do not accept a context as the first parameter.
+
+```typescript
```

Comment:
Would benefit from having an example here to show how to define a custom decorator.

Reply:
This needs to be tried before we'd have that.  I think we deprioritized custom event receivers.

Reply:
Agreed, let's deprioritize it. Maybe even cut this "Extensibility -> Decorators" section for now.